### PR TITLE
Similar typedefs for parametric shape

### PR DESCRIPTION
**Pull request by akrah**
# PR Description

This PR:
- homogenizes typdefs of parametric shapes that could be Points/Points2D and Vector/Vector2D
- modify constructors/destructors by using delete/default directives
- adds copy constructors that were strangely forbidden

This PR also:
- fixes computation of bounding boxes for some shapes
- removes some useless temporary variables
- improves comparisons to 0 by using isAlmostEqual method
- adds const directives to several method parameters

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

**Comment by copyme** [comment]
@akrah Cool :-) Are you using these parametric curves, btw?

**Comment by akrah** [comment]
Yes, I prepare an IPOL demo that evaluate discrete estimators on all
digitizations of an object, up to a translation. Of course, this demo
is based on DGtal ;), and mainly uses shapes and estimators.

**Comment by copyme** [comment]
@akrah cool! I thought that these are mainly used for examples and demos, though. So, I was a bit surprised to see PR regarding them :-)

**Comment by akrah** [comment]
@copyme I suppose it was probably the case because PRs I make are mostly about difficulties to homogenize shapes and estimators concepts in order to use generalization in code using them.

**Comment by copyme** [comment]
@akrah I agree. During my masters Yukiko asked us to write some experiments in DGtal concerning estimators. I wanted to have a very automatic and generic code and ended up with many adapters, etc. Unifying this is a marvelous idea, at least from a student stand point :D

**Comment by dcoeurjo** [comment]
Thx @akrah , could tou please add a Changelog entry ?

**Comment by akrah** [comment]
Done @dcoeurjo!

File: ChangeLog.md [hunk 1]
```diff
@@ -56,6 +56,9 @@
   - Fix Lemniscate definition following Bernoulli's definition
    (Adrien Krähenbühl,
    [#1427](https://github.com/DGtal-team/DGtal/pull/1427))
+  - Homogenizes typedefs of all parametric shapes and fixes some bounding box
+    computing (Adrien Krähenbühl,
```
Comment (dcoeurjo) [inline ChangeLog.md]:
```suggestion
  - Homogenizes typedefs of all parametric shapes and fixes some bounding box
    computations (Adrien Krähenbühl,
```

**Comment by dcoeurjo** [comment]
thanks @akrah for the cleanup !
all good for me

**Pull request merged**